### PR TITLE
[Fix-Proxy] Fix in memory caching memory leak 

### PR DESCRIPTION
**Pull request by ishaan-jaff**
## Type

🆕 New Feature
🐛 Bug Fix
🧹 Refactoring
📖 Documentation
🚄 Infrastructure
✅ Test

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
If UI changes, send a screenshot/GIF of working UI fixes

### CODE REVIEW

**Comment by Manouchehri** [comment]
Hmm, should we consider eventually switching to TLRUCache from cachetools? https://cachetools.readthedocs.io/en/latest/#cachetools.TLRUCache

I use it in an unrelated closed-source FastAPI application and it works great.

**Comment by ishaan-jaff** [comment]
Great suggestion @Manouchehri i think we can do it in this PR itself

File: litellm/caching.py [hunk 1]
```diff
@@ -139,8 +152,23 @@ async def async_increment(self, key, value: float, **kwargs) -> float:
...
         return value
 
+    async def clean_up_in_memory_cache(self):
+        """
+        Runs periodically to clean up the in-memory cache
+
+        - loop through all keys in cache, check if they are expired
+        - if yes, delete them
+        """
+        self.cache_dict = {}
```
Comment (jeromeroussin) [inline litellm/caching.py]: I am surprised you don't try to keep some entries around. Maybe I misunderstand the code, I imagined you'd need to keep entries for the current minute at least.

File: litellm/caching.py [hunk 2]
```diff
@@ -64,10 +64,16 @@ async def disconnect(self):
 
 
 class InMemoryCache(BaseCache):
```
Comment (jeromeroussin) [inline litellm/caching.py:66]: Does the cache in Redis keep growing and need regular cleanup?

Comment (ishaan-jaff) [reply litellm/caching.py:66]: yes it does - perhaps we can set a default ttl on redis caching. Planning on addressing that in a separate PR

File: litellm/caching.py (hunk 1 above)
Comment (ishaan-jaff) [reply litellm/caching.py]: fixed to use a custom eviction policy 

```python
def evict_cache(self):
        """
        Eviction policy:
        - check if any items in ttl_dict are expired -> remove them from ttl_dict and cache_dict
        This guarantees the following:
        - 1. When item ttl not set: At minimum each item will remain in memory for 10 minutes
        - 2. When ttl is set: the item will remain in memory for at least that amount of time specified
        - 3. the size of in-memory cache is bounded
        """
        for key in list(self.ttl_dict.keys()):
            if time.time() > self.ttl_dict[key]:
                self.cache_dict.pop(key, None)
                self.ttl_dict.pop(key, None)
```

**Pull request merged**